pkg/hooks/http/http.go

```diff
@@ -66,7 +81,16 @@ func (h HttpHook) InvokeHook(hookReq hooks.HookRequest) (hookRes hooks.HookRespo
 	}
 	defer httpRes.Body.Close()
 
-	httpBody, err := io.ReadAll(httpRes.Body)
+	contentType := httpRes.Header.Get("Content-Type")
+
+	if contentType != "" {
+		mediaType, _, err := mime.ParseMediaType(contentType)
+		if err != nil || mediaType != "application/json" {
+			return hookRes, fmt.Errorf("unexpected Content-Type: %s", contentType)
+		}
+	}
+
+	httpBody, err := io.ReadAll(io.LimitReader(httpRes.Body, h.SizeLimit))
```

Comment:
As mentioned in https://github.com/tus/tusd/issues/1273#issuecomment-2929286252, I would like to have an explicit error if the size limit was reached. This allows developers to immediately know that the problem was caused by the response exceeding the size.

The easiest approach to do so is to limit the reader to `h.SizeLimit+1` and then error out if more than `h.SizeLimit` was read.